container-images/intel-gpu/Containerfile

```diff
@@ -0,0 +1,32 @@
+FROM quay.io/fedora/fedora:41 as builder
+
+COPY intel-gpu/oneAPI.repo /etc/yum.repos.d/
+
+RUN dnf install -y lspci clinfo intel-opencl g++ cmake git libcurl-devel intel-oneapi-base-toolkit ; \
+    git clone https://github.com/ggerganov/llama.cpp.git -b b4523 ; \
+    cd llama.cpp ; \
+    mkdir -p build ; \
+    cd build ; \
+    source /opt/intel/oneapi/setvars.sh ; \
+    cmake .. -DGGML_SYCL=ON -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx -DLLAMA_CURL=ON -DGGML_CCACHE=OFF -DGGML_NATIVE=ON ; \
```

Comment:
We actually want DGGML_NATIVE=OFF

=ON creates not so portable binaries (and you get different results based on what generation CPU one has.

Reply:
Do we need another PR?

Reply:
Good point...  AFAIK, intermediate images are not automatically cleaned up.  At least not with local workstation builds.  Not sure how GitHub actions handles it though...

Reply:
AFAIK that's not an image, just a named layer, and whether it's cleaned afterwards, depends on whether image build has (layer) caching enabled or not...

Reply:
If one would want to avoid _both_ caching OneAPI build layer due to its disk usage, and re-downloads of layer content (when it's not cached), I would suggest:
* Using OneAPI base images instead of their package versions: https://hub.docker.com/r/intel/oneapi
* Doing `docker pull` to make sure latest version of the image(s) are present, and
* Running `docker build` with caching disabled.

Reply:
Another option for reducing layer size would be installing just suitable subset of `intel-oneapi-base-toolkit` metapackage dependencies directly.  Finding out which of them are not needed will need some iteration though...

(To test that, I would recommend running container with full `intel-oneapi-base-toolkit`, where subset of its packages is  uninstalled before build.  That way one avoid large deps being re-downloaded.)